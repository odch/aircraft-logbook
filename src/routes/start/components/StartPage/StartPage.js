import React from 'react'
import { Redirect } from 'react-router-dom'
import LoadingIcon from '../../../../components/LoadingIcon'
import { organization } from '../../../../shapes'
import isLoaded from '../../../../util/isLoaded'

class StartPage extends React.Component {
  render() {
    const { selectedOrganization } = this.props
    if (!isLoaded(selectedOrganization)) {
      return <LoadingIcon />
    }
    if (selectedOrganization) {
      return <Redirect to={`/organizations/${selectedOrganization.id}`} />
    }
    return <Redirect to="/organizations" />
  }
}

StartPage.propTypes = {
  selectedOrganization: organization
}

export default StartPage
