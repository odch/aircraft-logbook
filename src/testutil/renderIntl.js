import React from 'react'
import renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'
import messages from '../messages'

const LOCALE = 'de'

const renderIntl = component =>
  renderer.create(
    <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
      {component}
    </IntlProvider>
  )

export default renderIntl
