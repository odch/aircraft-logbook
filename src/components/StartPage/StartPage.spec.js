import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderIntl from '../../testutil/renderIntl'
import StartPage from './StartPage'

describe('components', () => {
  describe('StartPage', () => {
    it('renders correctly', () => {
      const store = configureStore()({
        firebase: {
          auth: {
            isEmpty: false,
            email: 'test@example.com'
          },
          profile: {}
        }
      })
      const tree = renderIntl(
        <Provider store={store}>
          <StartPage />
        </Provider>
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
