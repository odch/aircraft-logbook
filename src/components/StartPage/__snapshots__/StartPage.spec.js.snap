// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components StartPage renders correctly 1`] = `
Array [
  <div
    data-cy="username"
  >
    test@example.com
  </div>,
  <button
    onClick={[Function]}
  >
    <span>
      Abmelden
    </span>
  </button>,
]
`;
